File: Jump Game.java
public class Solution {
    public boolean canJump(int[] A) {
        int farest = 0;
        for (int i = 0; i < A.length; i++) {
            if (farest < i)
                return false;
            farest = Math.max(farest, i + A[i]);
            if (farest >= A.length - 1)
                break;
        }
        return true;
    }
}

File: Jump Game.py
class Solution:
	# @param A, a list of integers
	# @return a boolean
	def canJump(self, A):
		farest = 0
		for i in xrange(len(A)):
			if (farest < i):
				return False
			farest = max(farest, i + A[i])
			if (farest >= len(A) - 1):
				return True
		return True